js/everything.js

```diff
@@ -206,11 +212,55 @@ var ZenPlayer = {
                     seconds: plyrPlayer.plyr.embed.getDuration(),
                     youtubeID: plyrPlayer.plyr.embed.getVideoData().video_id
                 });
-
                 // Show player
                 that.show();
                 updateTweetMessage();
             });
+            // when player has finished playing
+            plyrPlayer.addEventListener("ended", function() {
+                plyrPlayer.removeEventListener("ended");
+                videoMetaData = window.sessionStorage.getItem("videoMetaData");
+                videoMetaData = JSON.parse(videoMetaData);
+                if (videoMetaData !== null) {
+                    videoMetaData["items"].push({"type" : "youtube", "id" : currentVideoID, "played" : "true" });
+                    videoMetaData["autoplayState"] = autoplayState ;
+                    if (autoplayState !== null && autoplayState === "true") {
+                        /*
+                        videoMetaData contains two variables, an autoplay state and information (list) about all videos played till now.
+                        It is stored in session storage, and retrieved when relevant entries need to be added/modified.
+                        */
+                        // if playList songs left
+                        if (playList.length !== 0)
```

Comment:
I'd like to see this if statement re-written, it's a bit difficult to follow, especially with the do/while loop. This is untested, but here's what I'm thinking:

```javascript
while (playList.length > 0) {
    var nextID = playList.pop();
    for (var i = 0; i < videoMetaData["items"].length; i++) {
        var curVideo = videoMetaData["items"][i];
        if (curVideo["id"] === nextID) {
            // restart the for loop
            nextID = playlist.pop();
            i = -1;
        }
    }
    that.playNext(nextID);
}
```